index.js

```diff
@@ -83,6 +84,10 @@ module.exports = function serialize(obj, options) {
             deleteFunctions(value);
         }
 
+        if (typeof value === 'number' && isNaN(value)) {
+            return '@__N-' + UID + '-' + (nans.push(value) - 1) + '__@';
+        }
```

Comment:
it might be better to combine this if statement with the `number` if statement.
https://github.com/yahoo/serialize-javascript/blob/5838b9199049ab253008280ccc4850bba9c8816e/index.js#L132-L134